### ISSUE TITLE
Fail CI if doxygen is missing

### ISSUE THREAD

**Issue by esoha-nvidia** [NVIDIA/cuCollections#519]
https://github.com/NVIDIA/cuCollections/blob/47766b33283e7646df14286481780abbe112fc11/ci/pre-commit/doxygen.sh#L20

Why would doxygen in CI be sometimes missing and sometimes not?  This check doesn't make sense to me...

Either change this to `exit 1` so that the test will fail or just delete the entire file.

If you really want to allow the test to skip when doxygen isn't found, let the caller write `./doxygen.sh || true`.

Failing to find doxygen silently is an error-prone default!

**Comment by sleeepyjack** [comment]
The problem with this script is that it is part of our `pre-commit` checks and is run in two different scenarios:

1. In local user setup: Users must pass all `pre-commit` checks to be able to push new commits upstream. Since the script requires doxygen to be locally installed in the right version and we don't want to put the burden on the developer to install the package before being able to contribute to cuco, we simply ignore the check by marking it as complete anyway (`exit 0`)
2. The same `pre-commit` checks are also run in CI. Here we probably want the script to `exit 1`, i.e., mark it as an error because we control the environment and can thus guarantee doxygen is installed in the right version.

So yeah, we need a way to distinguish between the local vs. CI environment to fix this issue properly.
An alternative (and maybe even better) solution would be to ship a suitable `doxygen` binary through `pre-commit` but AFAIK this isn't possible.

**Comment by esoha-nvidia** [comment]
It's working okay?

https://results.pre-commit.ci/run/github/226973898/1719338061.Rl73OP_9QtGpHF4sahHmPQ

**Comment by sleeepyjack** [comment]
Nah, it shouldn't be skipped in CI. That's what we need to fix.

**Comment by sleeepyjack** [comment]
I was wrong. Based on offline discussions, we cannot install doxygen in the `pre-commit` CI runner. Thus we won't enforce running the test.

Instead, we have a separate CI job with an environment that we can control and has the right doxygen version installed. This is then used to run `doxygen.sh` to verify docs.

tl;dr `exit 0` is the right strategy at the moment.
